separate vgpu and gpu passthrough samples (#180)
NVIDIA/cloud-native-docs: openshift/openshift-virtualization.rst

* If planning to use NVIDIA vGPU, SR-IOV must be enabled in the BIOS if your GPUs are based on the NVIDIA Ampere architecture or later. Refer to the `NVIDIA vGPU Documentation <https://docs.nvidia.com/grid/latest/grid-vgpu-user-guide/index.html#prereqs-vgpu>`_ to ensure you have met all of the prerequisites for using NVIDIA vGPU.

***********************************************************
Configure NVIDIA GPU Operator with OpenShift Virtualization
***********************************************************

After configuring the :ref:`prerequisites<prerequisites>`, the high level workflow for using the NVIDIA GPU Operator with OpenShift Virtualization is as follows:

> * :ref:`Enable the IOMMU driver <enable-iommu-driver>`.

Author: @cdesiniotis Is this step always required? Or only in certain cases? Just wanted to confirm before adding it here

   .. code-block:: console

      $ oc get node cnt-server-2 -o json | jq '.status.allocatable | with_entries(select(.key | startswith("nvidia.com/"))) | with_entries(select(.value != "0"))'

   *Example Output*

   .. code-blocK:: output

      {
        "nvidia.com/GA102GL_A10": "1"
      }

#. Determine the PCI device IDs for the GPUs.

   * You can search by device name in the `PCI IDs database <https://pci-ids.ucw.cz/v2.2/pci.ids>`_.

   * If you have host access to the node, you can list the NVIDIA GPU devices with a command like the following example:

     .. code-block:: console

        $ lspci -nnk -d 10de:

     *Example Output*

     .. code-block:: output
        :emphasize-lines: 1

        65:00.0 3D controller [0302]: NVIDIA Corporation GA102GL [A10] [10de:2236] (rev a1)
                Subsystem: NVIDIA Corporation GA102GL [A10] [10de:1482]
                Kernel modules: nvidiafb, nouveau

#. Modify the ``HyperConverged`` custom resource like the following partial examples.

   .. code-block:: yaml

      ...
      spec:
         featureGates:
            disableMDevConfiguration: true
         permittedHostDevices: # Defines VM devices to import.

>             pciHostDevices: # Include for GPU passthrough
>             - externalResourceProvider: true
>             pciDeviceSelector: 10DE:2236
>             resourceName: nvidia.com/GA102GL_A10

Reviewer: ```suggestion
            pciHostDevices: # Include for GPU passthrough
            - externalResourceProvider: true
              pciDeviceSelector: 10DE:2236
              resourceName: nvidia.com/GA102GL_A10
```
Should look like this, please fix the indentation.

      ...

>    Replace the values in the YAML as follows:
> 
>    * ``mdevNameSelector`` and ``resourceName`` under ``mediatedDevices`` to correspond to your vGPU type.

Reviewer: In this section we are referring to GPU Passthrough, so this should look like

```suggestion
   Replace the values in the YAML as follows:

   *  ``pciDeviceSelector`` and ``resourceName`` under ``pciHostDevices`` to correspond to your GPU model.
```

          - externalResourceProvider: true
            pciDeviceSelector: 10DE:2236
            resourceName: nvidia.com/GA102GL_A10
          mediatedDevices: # Include for vGPU
          - externalResourceProvider: true
            mdevNameSelector: NVIDIA A10-24Q

>             resourceName: nvidia.com/NVIDIA_A10-24Q

Reviewer 2: The yaml snippet here is incorrect. We need to keep the mediatedDevices section instea of the pciHostDevices section. In addition, as mentioned in another comment let's specify the `A10-12Q` profile for consistency.

   * Include ``permittedHostDevices`` for GPU passthrough. 

>      Include ``mediatedDevices`` for vGPU. Include boht sections if you are using GPU passthrough and vGPU

Reviewer 3: ```suggestion
     Include ``mediatedDevices`` for vGPU. Include both sections if you are using GPU passthrough and vGPU
```

        featureGates:
          disableMDevConfiguration: true
        permittedHostDevices: # Defines VM devices to import.
          mediatedDevices: # Include for vGPU
          - externalResourceProvider: true
            mdevNameSelector: NVIDIA A10-14Q

>             resourceName: nvidia.com/NVIDIA_A10-24Q

Reviewer: For consistent vGPU profile

```suggestion
            resourceName: nvidia.com/NVIDIA_A10-14Q
```

* The GPU devices are configured in the ``HyperConverged`` custom resource (CR).

Procedure
=========

#. Assign the GPU devices to a virtual machine (VM) by editing the ``spec.domain.devices.gpus`` field of the ``VirtualMachine`` manifest:

   Example for GPU passthrough:

   .. code-block:: yaml

      apiVersion: kubevirt.io/v1alpha3
      kind: VirtualMachineInstance
      ...
      spec:
      domain:
         devices:
            gpus:
            - deviceName: nvidia.com/GA102GL_A10
            name: gpu1
      ...

   Example for vGPU:

   .. code-block:: yaml
      
      apiVersion: kubevirt.io/v1alpha3
      kind: VirtualMachineInstance
      ...
      spec:
      domain:
         devices:
            gpus:

>             - deviceName: nvidia.com/NVIDIA_A10-12Q

Reviewer: ```suggestion
            - deviceName: nvidia.com/NVIDIA_A10-14Q
```
